mlir/lib/Transforms/Utils/DialectConversion.cpp

```diff
@@ -144,6 +144,8 @@ struct ConversionValueMapping {
   template <typename OldVal, typename NewVal>
   std::enable_if_t<IsValueVector<OldVal>::value && IsValueVector<NewVal>::value>
   map(OldVal &&oldVal, NewVal &&newVal) {
+    assert(mapping.find(oldVal) == mapping.end() &&
+           "attempting to overwrite mapping");
```

Comment:
It seems as if this assert will be user facing (eg if users accidently replace a value more than once?). Could we rephrase the assert message such that it makes more sense from an outside perspective, maybe something like 
> Cannot replace value more than once
or similar